posts/2024-05-01-spring-boot-3.adoc

```diff
@@ -0,0 +1,238 @@
+
+In this blog post, we'll first create a simple app by using the link:https://start.spring.io/[Spring Initializr], and then deploy it as a WAR to Liberty without any modifications, using the same deployment as for any WAR.
+
+We then show how to optimize the Spring Boot deployment by adding the Liberty Maven Plugin (`liberty-maven-plugin`) and a simple `server.xml` configuration with the `springBoot-3.0` feature.   We can then build an efficiently-layered Docker image.
+
+
+== Prerequisites
+- The sample requires Java 21 (though it could be easily reworked using Java 17)
+
+== Deploy as standard WAR
+
+1. Create a project from your browser using link:https://start.spring.io/#!type=maven-project&language=java&platformVersion=3.2.5&packaging=war&jvmVersion=21&groupId=demo&artifactId=spring-liberty&name=spring-liberty&description=Demo%20project%20for%20Spring%20Boot&packageName=demo&dependencies=web[this Spring Initializr configuration].
++
+2. Add the following Spring web controller file (use the `vi` command here, or your preferred editor) :
++
+[source,sh]
+----
+vi src/main/java/demo/HelloController.java
+----
++
+[source,java]
+----
+package demo;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HelloController {
+
+    @GetMapping("/")
+    public String hello() {
+        return "Hello from Spring Boot in Open Liberty!";
+    }
+
+}
+----
++
+3. From the command line, run a standard WAR deployment in https://openliberty.io/docs/latest/development-mode.html[dev mode], with the configuration provided by the `webProfile10` template.
++
+[source,sh]
+----
+./mvnw io.openliberty.tools:liberty-maven-plugin:3.10:dev -Dtemplate=webProfile10 -DskipTests
+----
++
+4. Test the application by visiting the following URL in a browser: http://localhost:9080/spring-liberty-0.0.1-SNAPSHOT/
++
+You should see:
++
+  Hello from Spring Boot in Open Liberty!
++
+5. Stop the server by pressing `Ctrl`+`C` or  by typing `q` and pressing `Enter` to quit dev mode.
+
+=== Results
+
+The app is deployed in dev mode, like you would with any Jakarta EE WAR, without any special configuration or steps specific to Spring Boot.   
+
+The Spring portion of the app is bootstrapped by the web container by using the `SpringBootServletInitializer` extension generated into your application by the **Spring Initializr** and from there you can use the Spring and Spring Boot APIs.
+
+== Use optimized Liberty Spring Boot deployment
+
+This example uses the same Spring app that is created in step 1 of the previous example. By adding some extra configuration in the `pom.xml` and `server.xml`, you can build an efficiently-layered Docker image and also parameterize the app with application arguments.
+
+1. Add liberty-maven-plugin with Spring Boot deployment config to the `pom.xml` file:
```

Comment:
```suggestion
1. Add the following `liberty-maven-plugin` with Spring Boot deployment config to the `pom.xml` file.
```

Reply:
That suggestion seemed a bit awkward to me.   While I'm not crazy about my next revision, I think the user will be clear enough what to do here, which is to copy and paste the XML snippet shown into the pom.xml.